airbyte-integrations/bases/base-java/src/main/resources/ssh-tunnel-spec.json

```diff
@@ -0,0 +1,164 @@
+{
+  "type": "object",
+  "title": "SSH Tunnel Method",
+  "description": "Whether to initiate an SSH tunnel before connecting to the database, and if so, which kind of authentication to use.",
+  "oneOf": [
+    {
+      "title": "No Tunnel",
+      "required": ["tunnel_method"],
+      "properties": {
+        "tunnel_method": {
+          "description": "No ssh tunnel needed to connect to database",
+          "type": "string",
+          "const": "NO_TUNNEL",
+          "order": 0
+        }
+      }
+    },
+    {
+      "title": "SSH Key Authentication",
+      "required": [
+        "tunnel_method",
+        "tunnel_host",
+        "tunnel_ssh_port",
+        "tunnel_username",
+        "tunnel_local_port",
+        "tunnel_user_ssh_key"
+      ],
+      "properties": {
+        "tunnel_method": {
+          "description": "Connect through a jump server tunnel host using username and ssh key",
+          "type": "string",
+          "const": "SSH_KEY_AUTH",
+          "order": 0
+        },
+        "tunnel_host": {
+          "title": "SSH Tunnel Jump Server Host",
+          "description": "Hostname of the jump server host that allows inbound ssh tunnel.",
+          "type": "string",
+          "order": 1
+        },
+        "tunnel_ssh_port": {
+          "title": "SSH Connection Port",
+          "description": "Port on the proxy/jump server that accepts inbound ssh connections.",
+          "type": "integer",
+          "minimum": 0,
+          "maximum": 65536,
+          "default": 22,
+          "examples": ["22"],
+          "order": 2
+        },
+        "tunnel_username": {
+          "title": "SSH Login Username",
+          "description": "OS-level username for logging into the jump server host.",
+          "type": "string",
+          "order": 3
+        },
+        "tunnel_user_ssh_key": {
+          "title": "SSH Private Key",
+          "description": "OS-level user account ssh key credentials for logging into the jump server host.",
+          "type": "string",
+          "airbyte_secret": true,
+          "multiline": true,
+          "order": 4
+        },
+        "tunnel_db_remote_host": {
```

Comment:
IMO this should be the current hostname field, no need to add an extra host field. 

Transparently we should replace the host field with `localhost` and use this in the SSH tunnel config instead.

Reply:
ah. good point. let me see what i need to do to make that work.

Reply:
@sherifnada - i followed the path here. it complicates the code a fair amount, but it does provide a better use experience. i think it is the right thing to do, but would love to get your eyes on it before I go forward. You can either look at this PR as it is now or if you want to see exactly what had to change see this commit: https://github.com/airbytehq/airbyte/pull/5742/commits/d55049d210e3a5a618d1b2164fa18301e5432738.

The thing that is a bit complicated is that you need to add logic that:
1. knows what fields in the config are the host and port
2. replaces those with localhost and the port that we use for the tunnel
3. make sure the source actually uses the new config

Because this is tricky, I went ahead and did the same SshSource decorator that I did for the destination. I think that at least contains all of the complexity in one spot.

Reply:
Nice job! I think you placed it in a good place